libbeat/otelcommon/oteltranslate/tls_otel.go

```diff
@@ -0,0 +1,137 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package oteltranslate
+
+import (
+	"crypto/tls"
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent-libs/transport/tlscommon"
+)
+
+// currently unsupported parameters
+// ssl.curve_types
+// ssl.ca_sha256
+// ssl.ca_trustred_fingerprint
+// ssl.key_passphrase
+
+// ssl.supported_protocols -> partially supported
+// ssl.restart_on_cert_change.*
+// ssl.renegotiation
+// ssl.verification_mode: All modes are not distinctly mapped yet
+func validateUnsupportedConfig(tlscfg *tlscommon.Config) error {
+	if len(tlscfg.CurveTypes) > 0 {
+		return errors.New("setting ssl.curve_types is currently is currently not supported")
+	}
+	if tlscfg.CATrustedFingerprint != "" {
+		return errors.New("setting ssl.ca_trusted_fingerprint is currently not supported")
+	}
+	if len(tlscfg.CASha256) > 0 {
+		return errors.New("setting ssl.ca_sha256 is currently not supported")
+	}
+	if tlscfg.Certificate.Passphrase != "" {
+		return errors.New("setting ssl.key_passphrase is currently not supported")
+	}
+	return nil
+}
+
+// TLSCommonToOTel converts a tlscommon.Config into the OTel configtls.ClientConfig
+func TLSCommonToOTel(tlscfg *tlscommon.Config) (map[string]any, error) {
```

Comment:
Shouldn't be better to return a configtls.ClientConfig explicitly? I mean, if the point of this is to translate the config properly I assume the type safety of having the real config struct can guard against unwanted changes and breaking changes during otel updates.

Reply:
the only reason we return `map[string]any` is for it to be compatible with the final elasticsearch output [here](https://github.com/elastic/beats/pull/41849/files#diff-3ed5a6680266f464de20f46c52b1d741b65d5b52532a4b20aecc2ae9f1df4018R128) 

We could return `configtls.ClientConfig` and then decode it into a map structure ->  but in that case, it also embeds empty fields on `configtls.ClientConfig`. Since this part of the code only performs basic translation for now, we can take it up as an improvement later? wdyt

Reply:
I had a discussion with Khushi on slack and we cannot return `configtls.ClientConfig` directly. 
The reason being that some of the members of `configtls.ClientConfig`  struct, for eg `KeyPem` has a custom type, [configopaque.String](https://github.com/open-telemetry/opentelemetry-collector/blob/52d8a1a367dd3c444624fab6b69e04eec69436a4/config/configopaque/opaque.go#L12-L36). 

Once we marshal the map, we cannot then unmarshal it or merge it with other configs because the "custom" string type redacts the data.

Reply:
I have an added explicit type safety check instead. https://github.com/elastic/beats/pull/41849/files#diff-c86a667df5829a587c4daa5d36ce9af5510a0e0b0fd9db0fcec8816e03f02d1eR138-R151